pkg/cryptoauthlib/contrib/atca.c

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2019 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_cryptoauthlib cryptoauthlib security crypto
+ * @{
+ *
+ * @file
+ * @brief       HAL implementation for the library Microchip CryptoAuth devices
+ *
+ * @author      Lena Boeckmann <lena.boeckmann@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include "xtimer.h"
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+#include "atca.h"
+#include "atca_params.h"
+
+
+/* Timer functions */
+void atca_delay_us(uint32_t delay)
+{
+    xtimer_usleep(delay);
+}
+
+void atca_delay_10us(uint32_t delay)
+{
+    xtimer_usleep(delay * 10);
+}
+
+void atca_delay_ms(uint32_t delay)
+{
+    xtimer_usleep(delay * 1000);
+}
+
+/* Hal I2C implementation */
+ATCA_STATUS hal_i2c_init(void *hal, ATCAIfaceCfg *cfg)
+{   
+    if(cfg->iface_type != ATCA_I2C_IFACE)
+    {
```

Comment:
Coding style: opening brace must *not* be on separate line, please check throughout the code